### PR TITLE
Organization Chart 2 - Modals and Dropdown menus to add/remove entities

### PR DESCRIPTION
**Pull request by FabioDiegoMastrorilli**
This is the second PR related to the Org Chart refactoring to be reviewed after the [first one](https://github.com/liferay-frontend/liferay-portal/pull/1311) gets merged.

Here I connected modals and dropdown menu to a d3 chart. in order to let users be able to create and remove connections among organizations, accounts and users.

**What this PR allows:**

- [x] navigating through organisations, accounts and users in a static environment
- [x] creating new organizations via Modal
- [x] creating or adding accounts within organizations
- [x] inviting users within accounts
- [x] removing / deleting entities via dropdown menu
- [ ] update entities connection via drag & drop
- [ ] multi selection via shift key
- [ ] paths highlighting at mouseover
- [ ] actions available when user has the right permissions
- [x] it works in portal
- [ ] it supports a table view
- [ ] it supports a map view

**Design**

Here's the final expected result: https://www.figma.com/file/Fgh6jD5qPBr8srVZlmxAWc/(m)-commerce-classic-theme-01a-ac?node-id=5037%3A95299. 

Please note: the chart won't work until [this PR](https://github.com/liferay-usm/liferay-portal/pull/189) gets merged

Thanks in advance,
Fabio

cc @nhpatt, @riccardo-alberti

### CODE REVIEW

**Comment by FabioDiegoMastrorilli** [comment]
ci:test:sf

File: modules/dxp/apps/commerce/commerce-organization-web/src/main/java/com/liferay/commerce/organization/web/internal/portlet/CommerceOrganizationPortlet.java [hunk 1]
```diff
@@ -71,53 +64,38 @@
...
+						_organizationService, _userLocalService);
 
-		renderRequest.setAttribute(
-			WebKeys.PORTLET_DISPLAY_CONTEXT,
-			commerceOrganizationDisplayContext);
+			renderRequest.setAttribute(
+				WebKeys.PORTLET_DISPLAY_CONTEXT,
+				commerceOrganizationDisplayContext);
+		}
+		catch (PortalException portalException) {
```
Comment (nhpatt) [inline modules/dxp/apps/commerce/commerce-organization-web/src/main/java/com/liferay/commerce/organization/web/internal/portlet/CommerceOrganizationPortlet.java:83]: You can throw exception and ignore the try catch if you want.

**Comment by markocikos** [comment]
This looks like work in progress, I'll convert it to draft.

File: modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/menu/MenuProvider.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	data: propData,
+	parentData,
+}) {
+	const [active, setActive] = useState(false);
+	const [data, updateData] = useState(false);
+	const menuRef = useRef(null);
+
+	/**
+	 * the useEffect below force the component repositioning
+	 */
```
Comment (kresimir-coko) [inline modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/menu/MenuProvider.js:36]: We don't want comments

File: modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/modals/AddAccountsModal.js [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+						updateAccountDetails(account.id, {
+							organizationIds: [
+								...account.organizationIds,
+								parentData.id,
+							],
+						})
+					)
+				).then((results) => {
+					const nodeChildren = [];
+					results.forEach((result) => {
```
Comment (kresimir-coko) [inline modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/modals/AddAccountsModal.js:109]: Empty line

File: modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/modals/AddAccountsModal.js [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+
+		setSelectedAccounts(filteredItems);
+	}
+
+	return (
+		<ClayModal observer={observer} size="md">
+			<ClayModal.Header>
+				{Liferay.Language.get('add-accounts')}
+			</ClayModal.Header>
+			<ClayModal.Body>
```
Comment (kresimir-coko) [inline modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/modals/AddAccountsModal.js:161]: Empty line between sibling elements

File: modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/modals/AddOrganizationsModal.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+	const [query, setQuery] = useState('');
+	const [items, setItems] = useState([]);
+	const [errors, setErrors] = useState([]);
+	const {chartInstanceRef} = useContext(ChartContext);
+
+	function handleSave() {
+		const newOrganizations = query
+			? items.concat({label: query, value: query})
+			: [...items];
+		const organizationNames = newOrganizations.map((item) => item.label);
```
Comment (kresimir-coko) [inline modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/modals/AddOrganizationsModal.js:38]: Empty line

File: modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/modals/AddOrganizationsModal.js [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+					newErrors.add(result.reason.title);
+				}
+				else {
+					newOrganizationsDetails.push(result.value);
+				}
+			});
+
+			setItems(failedOrganizations);
+			setErrors(Array.from(newErrors));
+			setQuery('');
```
Comment (kresimir-coko) [inline modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/modals/AddOrganizationsModal.js:65]: I think that we want empty lines between lines of code that execute stuff (anything except variable declarations)

File: modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/modals/AddOrganizationsModal.js [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+			}
+		});
+	}
+
+	return (
+		<ClayModal center observer={observer} size="md">
+			<ClayModal.Header>
+				{Liferay.Language.get('add-organizations')}
+			</ClayModal.Header>
+			<ClayModal.Body>
```
Comment (kresimir-coko) [inline modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/modals/AddOrganizationsModal.js:102]: Empty line between sibling elements

**Review by kresimir-coko — CHANGES_REQUESTED** [review]
There's a quintillion changes here, so I went through most and I'm seeing more repeats of same mistakes as in PRs from more than a month ago.

Please go through them and fix them, noting not just the ones I pointed out, but going over all the files.

File: modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/menu/MenuProvider.js (hunk 2 above)
Comment (bryceosterhaus) [reply modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/menu/MenuProvider.js:36]: @kresimir-coko this sort of comment should be okay, as it seems to be explaining why we would be using this useEffect. I could imagine someone refactoring this in the future an accidentally causing a regressing if they didn't know why there were two separate effects.

[Per Guidelines](https://github.com/liferay/liferay-frontend-projects/blob/master/guidelines/general/comments.md#guidelines)
> For cases where you want to explain some internal detail of the code (to help the reader understand how it works, but where such explanation wouldn't really belong in any extracted "documentation")

Comment (FabioDiegoMastrorilli) [reply modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/menu/MenuProvider.js:36]: I'll update it in a single line comment starting with capital letter ;)

File: modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/modals/AddOrganizationsModal.js (hunk 6 above)
Comment (FabioDiegoMastrorilli) [reply modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/modals/AddOrganizationsModal.js:66]: I would keep it this way for now. Will be changed in a next PR when this code will be updated anyway not to generate conflicts

**Comment by FabioDiegoMastrorilli** [comment]
ci:test:sf

**Comment by FabioDiegoMastrorilli** [comment]
ci:forward

**Pull request closed without merging**